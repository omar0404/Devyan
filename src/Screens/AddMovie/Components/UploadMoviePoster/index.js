import UploadMoviePoster from './UploadMoviePoster';
export default UploadMoviePoster;
