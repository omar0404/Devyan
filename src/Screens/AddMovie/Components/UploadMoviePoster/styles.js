import {StyleSheet} from 'react-native';
import theme from 'MoviesApp/src/Theme';
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexBasis: 40,
    borderWidth: 1,
    borderColor: 'gray',
    marginRight: 20,
    borderStyle: 'dotted',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: theme.text.secondary,
    textAlign: 'center',
    marginTop: 12,
  },
  poster: {
    width: '100%',
    flex: 1,
  },
});
export default styles;
