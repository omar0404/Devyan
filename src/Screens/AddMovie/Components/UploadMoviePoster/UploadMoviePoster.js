import React, {useState} from 'react';
import {launchImageLibrary} from 'react-native-image-picker';
import {TouchableOpacity, Text, Image} from 'react-native';
import styles from './styles';
import Entypo from 'react-native-vector-icons/Entypo';
import {isFunction as _isFunction} from 'lodash';
import PropTypes from 'prop-types';
const UploadMoviePoster = ({onImageUpload}) => {
  const [imageUri, setImageUri] = useState(null);
  const handleUploadMoviePoster = () => {
    const options = {
      noData: false,
    };
    launchImageLibrary(options, (response) => {
      setImageUri(response.uri);
      _isFunction(onImageUpload) && onImageUpload(response.uri);
    });
  };
  return (
    <TouchableOpacity
      onPress={handleUploadMoviePoster}
      style={styles.container}>
      {imageUri ? (
        <Image
          resizeMode={'cover'}
          source={{
            uri: imageUri,
          }}
          style={styles.poster}
        />
      ) : (
        <>
          <Entypo name={'image'} color={'gray'} size={50} />
          <Text style={styles.text}> Upload Movie{'\n'}Poster</Text>
        </>
      )}
    </TouchableOpacity>
  );
};
UploadMoviePoster.propTypes = {
  onImageUpload: PropTypes.func,
};
export default UploadMoviePoster;
