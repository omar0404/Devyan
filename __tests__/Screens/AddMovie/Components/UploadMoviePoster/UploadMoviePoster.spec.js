import React from 'react';
import UploadMoviePoster from 'MoviesApp/src/Screens/AddMovie/Components/UploadMoviePoster';
import {shallow} from 'enzyme';
import {TouchableOpacity} from 'react-native';

describe('UploadMoviePoster Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<UploadMoviePoster onImageUpload={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find(TouchableOpacity).prop('onPress')();
  });
});
