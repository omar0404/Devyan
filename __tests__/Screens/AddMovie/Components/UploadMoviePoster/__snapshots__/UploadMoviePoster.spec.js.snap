// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadMoviePoster Component should render correctly 1`] = `ShallowWrapper {}`;
